processor/src/main/java/com/log/processor/core/LogItemBuilderImpl.java

```diff
@@ -79,6 +107,45 @@ private String readResultLine(List<String> list, LogItem item) {
 		return lineStr;
 	}
 	
+	private void buildLogItemAttributesForS3(S3Api s3Api, AmazonS3 s3, String lineStr, String bucketName, String key, LogItem item) {
+		
+		int index = key.indexOf("/");
+		String s = key.substring(0, index);
+		String testElements[] = s.split("_");
```

Comment:
define all constants like "_" in separate class as static final and use them here. It is is easy to maintain the code